File: 150-evaluate-reverse-polish-notation/150-evaluate-reverse-polish-notation.java
class Solution {
    public int evalRPN(String[] tokens) {
        if(tokens.length == 1) return Integer.valueOf(tokens[0]);
        
        Stack<Integer> st = new Stack();
        int temp = 0;
        
        for(String s:tokens) {
            switch(s) {
                case "+": 
                    temp = st.pop();
                    temp += st.pop();
                    st.push(temp);
                    break;
                case "-":
                    temp = st.pop();
                    temp = st.pop() - temp;
                    st.push(temp);
                    break;
                case "*":
                    temp = st.pop();
                    temp *= st.pop();
                    st.push(temp);
                    break;
                case "/":
                    temp = st.pop();
                    temp = st.pop() / temp;
                    st.push(temp);
                    break;
                default:
                    st.push(Integer.valueOf(s));
                    break;
            }
        }
        
        return st.pop();
    }
}

File: 150-evaluate-reverse-polish-notation/150-evaluate-reverse-polish-notation.py
class Solution:
    def evalRPN(self, tokens: List[str]) -> int:
        if len(tokens) == 1:
            return int(tokens[0])
        
        stack = []
        
        for s in tokens:
            if s == "+":
                a = stack.pop()
                b = stack.pop()
                stack.append(b+a)
            elif s == "-":
                a = stack.pop()
                b = stack.pop()
                stack.append(b-a)
            elif s == "*":
                a = stack.pop()
                b = stack.pop()
                stack.append(b*a)
            elif s == "/":
                a = stack.pop()
                b = stack.pop()
                stack.append(int(b/a))
            else:
                stack.append(int(s))
        
        return stack.pop()